contracts/main/stablecoin-core/LiquidationEngine.sol

```diff
@@ -107,11 +107,26 @@ contract LiquidationEngine is PausableUpgradeable, ReentrancyGuardUpgradeable, I
     address[] calldata _collateralRecipients,
     bytes[] calldata datas
   ) external override nonReentrant whenNotPaused onlyWhitelisted {
+
+    require(_collateralPoolIds.length == _positionAddresses.length &&
+            _collateralPoolIds.length == _debtShareToBeLiquidateds.length && 
+            _collateralPoolIds.length == _maxDebtShareToBeLiquidateds.length && 
+            _collateralPoolIds.length == _collateralRecipients.length && 
+            _collateralPoolIds.length == datas.length 
+             ,"LiquidationEngine/batchLiquidate-invalid-arguments");
+
     for(uint i = 0; i < _collateralPoolIds.length; i++){
         try this.liquidate(_collateralPoolIds[i], _positionAddresses[i],_debtShareToBeLiquidateds[i], _maxDebtShareToBeLiquidateds[i], _collateralRecipients[i], datas[i],msg.sender){
         } catch Error(string memory reason) {
             emit LiquidationFail(_collateralPoolIds[i], _positionAddresses[i], msg.sender, reason);
             continue;
+        } catch Panic(uint errorCode) {
```

Comment:
Will look up quickly if there is a lazy lazy way to uint to stirng if not, we can just pass